src/common/src/cast/mod.rs

```diff
@@ -160,6 +176,40 @@ pub fn i64_to_timestamp(t: i64) -> Result<Timestamp> {
     ))
 }
 
+/// Converts str to interval for postgres debezium cdc json
+///
+/// The input str must have the following format:
+/// P<years>Y<months>M<days>DT<hours>H<minutes>M<seconds>S
+///
+/// Example
+/// - P1Y2M3DT4H5M6.78S
+#[inline]
+pub fn str_to_interval(s: &str) -> Result<Interval> {
+    let re = Regex::new(r"P([0-9]+)Y([0-9]+)M([0-9]+)DT([0-9]+)H([0-9]+)M([0-9]+(?:\.[0-9]+)?)S")
+        .unwrap();
+    let caps = re
+        .captures(s)
+        .ok_or(PARSE_ERROR_STR_TO_INTERVAL.to_string())?;
+    // safe to unwrap, since capture group is `Some`
+    let years: i32 = caps.get(1).unwrap().as_str().parse().unwrap();
+    let months: i32 = caps.get(2).unwrap().as_str().parse().unwrap();
+    let days = caps.get(3).unwrap().as_str().parse().unwrap();
+    let hours: i64 = caps.get(4).unwrap().as_str().parse().unwrap();
+    let minutes: i64 = caps.get(5).unwrap().as_str().parse().unwrap();
+    // usecs = sec * 1000000, use decimal to be exact
+    let usecs: i64 = (Decimal::from_str_exact(caps.get(5).unwrap().as_str())
+        .map_err(|_| PARSE_ERROR_STR_TO_INTERVAL.to_string())?
+        * Decimal::from_str_exact("1000000").unwrap())
+    .to_string()
+    .parse()
+    .unwrap();
```

Comment:
* No need to `.to_string().parse()`. Just `.try_into()`.
* Handle all errors rather than `unwrap()`. Use `checked_mul` rather than `*`.